test/README.md

```diff
@@ -0,0 +1,56 @@
+# Test
+
+This directory contains tests and testing docs for `Kn client`:
+
+- [Unit tests](#running-unit-tests) currently reside in the codebase alongside
+  the code they test
+- [End-to-end tests](#running-end-to-end-tests) :
+  - end-to-end tests in [`/test/e2e`](./e2e)
+
+## Running unit tests
+
+To run all unit tests:
+
+```bash
+$ hack/build.sh -f -t
+```
+
+_By default `go test` will not run [the e2e tests](#running-end-to-end-tests-locally),
+which need [`-tags=e2e`](#running-end-to-end-tests) to be enabled._
+
+## Running end to end tests locally
+
+To run [the e2e tests](./e2e) , you need to have a 
+
+1. [running knative environment.](./../DEVELOPMENT.md#create-a-cluster)
+2. kn binary in the $path. 
+
+Before running the e2e tests please make sure you dont have any namespaces with the name starting with `kne2etests`
+
+```bash
+$ ./e2e-tests-local.sh
+```
+
```

Comment:
please add running e2e tests selectively using tags as mentioned in the PR description at https://github.com/knative/client/pull/547#issue-351756917